src/localization/online_localizer/online_localizer.cpp

```diff
@@ -422,9 +370,8 @@ bool OnlineLocalizer::isLost(int N, double perc) const {
       lostFactor++;
     }
   }
-
-  if ((double)lostFactor / path.size() > perc) {
-    printf("[INFO][OnlineLocalizer] LOST localization\n");
+  if ((double)lostFactor / path.size() > ratio) {
```

Comment:
Don't use the C-style cast as it might use a different cast from what you expect. Use `static_cast<double>(lostFactor)` here instead to guarantee it is using the static cast.